joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/AstCreator.scala

```diff
@@ -331,19 +332,29 @@ class AstCreator(fileWithMeta: KtFileWithMeta, xTypeInfoProvider: TypeInfoProvid
   def astsForDeclaration(decl: KtDeclaration)(implicit typeInfoProvider: TypeInfoProvider): Seq[Ast] = {
     debugScope.push(decl)
     val result =
-      decl match {
-        case c: KtClass             => astsForClassOrObject(c)
-        case o: KtObjectDeclaration => astsForClassOrObject(o)
-        case n: KtNamedFunction =>
-          val isExtensionFn = typeInfoProvider.isExtensionFn(n)
-          astsForMethod(n, isExtensionFn)
-        case t: KtTypeAlias            => Seq(astForTypeAlias(t))
-        case s: KtSecondaryConstructor => Seq(astForUnknown(s, None, None))
-        case p: KtProperty             => astsForProperty(p)
-        case unhandled =>
-          logger.error(
-            s"Unknown declaration type encountered with text `${unhandled.getText}` and class `${unhandled.getClass}`!"
-          )
+      try {
+        decl match {
+          case c: KtClass             => astsForClassOrObject(c)
+          case o: KtObjectDeclaration => astsForClassOrObject(o)
+          case n: KtNamedFunction =>
+            val isExtensionFn = typeInfoProvider.isExtensionFn(n)
+            astsForMethod(n, isExtensionFn)
+          case t: KtTypeAlias            => Seq(astForTypeAlias(t))
+          case s: KtSecondaryConstructor => Seq(astForUnknown(s, None, None))
+          case p: KtProperty             => astsForProperty(p)
+          case unhandled =>
+            logger.error(
+              s"Unknown declaration type encountered with text `${unhandled.getText}` and class `${unhandled.getClass}`!"
+            )
+            Seq()
+        }
+      } catch {
+        case exception: Exception =>
```

Comment:
`Exception` is very broad - might want to use `scala.util.control.NonFatal` here for something slightly stricter